release/pipline/dags/istio_common_dag.py

```diff
@@ -0,0 +1,251 @@
+"""Airfow DAG and helpers used in one or more istio release pipline."""
+
+"""Copyright 2017 Istio Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+import datetime
+import time
+
+from airflow import DAG
+from airflow.models import Variable
+from airflow.operators.bash_operator import BashOperator
+from airflow.operators.dummy_operator import DummyOperator
+from airflow.operators.python_operator import BranchPythonOperator
+from airflow.operators.python_operator import PythonOperator
+
+import environment_config
+from gcs_to_gcs import GoogleCloudStorageToGoogleCloudStorageOperator
+
+YESTERDAY = datetime.datetime.combine(
+    datetime.datetime.today() - datetime.timedelta(days=1),
+    datetime.datetime.min.time())
+
+default_args = {
+    'owner': 'laane',
+    'depends_on_past': False,
+    'start_date': YESTERDAY,
+    'email': ['laane@google.com'],
+    'email_on_failure': False,
+    'email_on_retry': False,
+    'retries': 1,
+    'retry_delay': datetime.timedelta(minutes=5),
+}
+
+
+def MakeCommonDag(name='istio_daily_flow_test',
+                  schedule_interval='15 3 * * *',
+                  monthly=False):
+  common_dag = DAG(
+      name,
+      default_args=default_args,
+      schedule_interval=schedule_interval,
+  )
+
+  def AirflowGetVariableOrBaseCase(var, base):
+    try:
+      return Variable.get(var)
+    except KeyError:
+      return base
+
+
+  def GenerateTestArgs(**kwargs):
+    conf = kwargs['dag_run'].conf
+    if conf is None:
+      conf = dict()
+
+    date = kwargs['execution_date']
+
+    timestamp = time.mktime(date.timetuple())
+
+    days_since_zero = (date - datetime.datetime(2017, 8, 1, 0, 0, 0, 0)).days
```

Comment:
Just curious, why do we want this as initial date? some comment? 
And put this constant somewhere else, instead of in the code? 
Same for the below magic number "30".